pkg/prometheus/statefulset.go

```diff
@@ -1080,3 +1089,54 @@ func queryLogFilePath(p *monitoringv1.Prometheus) string {
 
 	return filepath.Join(defaultQueryLogDirectory, p.Spec.QueryLogFile)
 }
+
+func intersection(a, b []string) (i []string) {
+	m := make(map[string]bool)
+
+	for _, item := range a {
+		m[item] = true
+	}
+
+	for _, item := range b {
+		if _, ok := m[item]; ok {
+			i = append(i, item)
+		}
+	}
+	return i
+}
+
+func extractArgKeys(args []monitoringv1.Argument) []string {
+	var k []string
+	for _, arg := range args {
+		key := arg.Name
+		k = append(k, key)
+	}
+
+	return k
+}
+
+func verifyAdditionalArgs(operatorArgs []monitoringv1.Argument, additionalArgs []monitoringv1.Argument) (e error) {
+	operatorArgKeys := extractArgKeys(operatorArgs)
+	additionalArgKeys := extractArgKeys(additionalArgs)
+
+	i := intersection(operatorArgKeys, additionalArgKeys)
```

Comment:
we need to take care of bool args like `--storage.tsdb.no-lockfile`/`--no-storage.tsdb.no-lockfile`.